gum/backend-linux/gumandroid.c

```diff
@@ -641,16 +644,44 @@ gum_try_init_linker_details (void)
   return result;
 }
 
+static gchar *
+gum_find_linker_path ()
+{
+  gchar *linker_expected_path, *linker_path = NULL;
+
+  if (sizeof (gpointer) == 4)
+    linker_expected_path = "/system/bin/linker";
+  else
+    linker_expected_path = "/system/bin/linker64";
+
+  // If there are no linker file, fail
+  if (!g_file_test (linker_expected_path, G_FILE_TEST_EXISTS))
+    return NULL;
+
+  if (g_file_test (linker_expected_path, G_FILE_TEST_IS_SYMLINK))
+  {
+    linker_path = g_file_read_link (linker_expected_path, NULL);
+
+    // Issue reading the link, fall back to backwards compatible path
+    if (linker_path != NULL)
+      return linker_path;
+  }
+
+  // If file was not a symlink or symlink resolution doesn't work, fall back
+  // to expected file
+  return linker_expected_path;
+}
```

Comment:
How about something like:

```C
static gchar *
gum_find_linker_path (void)
{
  gchar * result = NULL;
  const gchar * traditional_path;

  traditional_path = (sizeof (gpointer) == 4)
      ? "/system/bin/linker"
      : "/system/bin/linker64";

  if (g_file_test (traditional_path, G_FILE_TEST_IS_SYMLINK))
    result = g_file_read_link (traditional_path, NULL);

  if (result == NULL)
    result = g_strdup (traditional_path);

  return result;
}
```

Basically:
- Always return a newly allocated string, not just in the symlink-case.
- Never return `NULL`, as it's not clear what we should do when the file doesn't exist (or the underlying `stat` fails so it looks like it doesn't) – so perhaps it's better to not deal with it here until we need a fallback strategy for that case.
- Shorten the variable names as the function now gives them context. (Now that they're no longer part of a larger function.)

Reply:
Works for me.